src/badges.js

```diff
@@ -0,0 +1,953 @@
+		return fileParsed;
+	} catch (err) {
+		console.log(chalk.red(`Error: invalid JSON in file: ${filePath}`));
+		return {};
+	}
+}
+
+export async function findJsonFiles(dir) {
+	// get all JSON files (exclude looking in excluded directories)
+	try {
+		const details = await fsp.stat(dir);
+		if (!details || !details.isDirectory) {
+			throw 'Directory not provided.';
+		}
+
+		let templateFiles = [];
+
+		const contents = await fsp.readdir(dir);
+		const readPromises = contents
+			.filter((file) => {
+				const filePath = path.resolve(dir, file);
+				try {
+					const fileStats = fs.statSync(filePath);
+
+					if (!fileStats.isSymbolicLink() && fileStats.isDirectory() && !DIR_EXCLUDE_LIST.includes(file)) {
+						return file;
+					} else if (file.match(/\.json$/)) {
+						templateFiles.push(filePath);
+					}
+				} catch (err) {
+					// not doing anything currently...
+				}
+			})
+			.map((file) => {
+				return findJsonFiles(path.resolve(dir, file));
+			});
+
+		const dirContents = await Promise.all(readPromises);
+
+		return [...templateFiles, ...dirContents.flat(1)];
+	} catch (err) {
+		console.log(chalk.red(`Error: cannot find templates in: ${dir}`));
+		throw err;
+	}
+}
+
+export function buildBadgeLocationsPayload(template) {
+	return {
+		type: template.type,
+		overlay: template.overlay,
+		callout: template.callout,
+	};
+}
+export function buildBadgeTemplatePayload(template) {
+	return {
+		type: template.type,
+		name: template.name,
+		label: template.label || `${pascalCase(template.name)} Badge`,
+		description: template.description || `${template.name} custom template`,
+		component: template.component,
+		locations: template.locations,
+		value: {
+			enabled: typeof template.value.enabled === 'boolean' ? template.value.enabled : true,
+			validations: template.value.validations || null,
+		},
+		parameters:
+			template.parameters?.map((parameter) => {
+				// mapping to only include the necessary fields
+				return {
+					name: parameter.name,
+					type: parameter.type,
+					label: parameter.label,
+					description: parameter.description,
+					defaultValue: parameter.defaultValue,
+					validations: parameter.validations,
+					options: parameter.options,
+				};
+			}) || [],
+	};
+}
+
+export function pascalCase(string) {
+	return `${string}`
+		.replace(new RegExp(/[-_]+/, 'g'), ' ')
+		.replace(new RegExp(/[^\w\s]/, 'g'), '')
+		.replace(new RegExp(/\s+(.)(\w*)/, 'g'), ($1, $2, $3) => `${$2.toUpperCase() + $3.toLowerCase()}`)
+		.replace(new RegExp(/\w/), (s) => s.toUpperCase());
+}
+
+export function handleize(input) {
+	if (typeof input != 'string') {
+		return input;
+	}
+
+	let handleized = input.toLowerCase();
+	handleized = handleized.replace(/[^\w\s]/g, '').trim();
+	handleized = handleized.replace(/\s/g, '-');
+	return handleized;
+}
+
+export async function timeout(microSeconds) {
+	return new Promise((resolve, reject) => {
+		setTimeout(resolve, microSeconds);
+	});
+}
```

Comment:
Looks like these are repeated helper functions (same found in recs), can we put them in utilities and share them?